packages/nuxt/src/app/nuxt.ts

```diff
@@ -28,7 +28,7 @@ type NuxtMeta = {
 
 type HookResult = Promise<void> | void
 
-type AppRenderedContext = { ssrContext: NuxtApp['ssrContext'] }
+type AppRenderedContext = { ssrContext: NuxtApp['ssrContext'], renderResult: Awaited<ReturnType<ReturnType<typeof createRenderer>['renderToString']>> }
```

Comment:
In https://github.com/nuxt/nuxt/pull/21412 the renderResult may be `null` if we are about to redirect immediately:

```suggestion
type AppRenderedContext = { ssrContext: NuxtApp['ssrContext'], renderResult: null | Awaited<ReturnType<ReturnType<typeof createRenderer>['renderToString']>> }
```